### PR TITLE
(enhancement): Extend get and update ruleset DQ methods

### PR DESCRIPTION
**Pull request by jaidisido**
### Feature or Bugfix

- Enhancement

### Detail
- `get_ruleset` should accept multiple rulesets and combine them to a data frame
- At the moment `update_ruleset` just overwrites the existing ruleset. Introducing a `mode` argument (overwrite, upsert) to handle upserts
- Wrap get methods in `try_it` to handle request throttling
- Add tests

### Relates
- <URL or Ticket>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awswrangler/data_quality/_create.py [hunk 1]
```diff
@@ -134,6 +136,8 @@ def update_ruleset(
         Ruleset name.
     updated_name : str
         New ruleset name if renaming an existing ruleset.
+    mode : str
+        overwrite (default) or upsert.
```
Comment (jaidisido) [inline awswrangler/data_quality/_create.py:140]: overwrite and upsert are the only modes I can think of

File: awswrangler/data_quality/_create.py [hunk 2]
```diff
@@ -145,18 +149,36 @@ def update_ruleset(
...
+    >>>     mode="insert",
+    >>>     dqdl_rules="Rules = [ RowCount between 1 and 3 ]",
+    >>>)
     """
     if (df_rules is not None and dqdl_rules) or (df_rules is None and not dqdl_rules):
         raise exceptions.InvalidArgumentCombination("You must pass either ruleset `df_rules` or `dqdl_rules`.")
+    if mode not in ["overwrite", "upsert"]:
+        raise exceptions.InvalidArgumentValue("`mode` must be one of 'overwrite' or 'upsert'.")
+
+    if mode == "upsert":
```
Comment (jaidisido) [inline awswrangler/data_quality/_create.py:171]: Could not find a better way to do an upsert in pandas

File: awswrangler/data_quality/_get.py [hunk 3]
```diff
@@ -9,28 +9,39 @@
...
+    >>> df_rulesets = wr.data_quality.get_ruleset(name=["ruleset_1", "ruleset_2"])
     """
-    rules = cast(str, _get_ruleset(ruleset_name=name, boto3_session=boto3_session)["Ruleset"])
-    return _rules_to_df(rules=rules)
+    ruleset_names: List[str] = name if isinstance(name, list) else [name]
+    dfs: List[pd.DataFrame] = []
+    for ruleset_name in ruleset_names:
+        rules = cast(str, _get_ruleset(ruleset_name=ruleset_name, boto3_session=boto3_session)["Ruleset"])
+        df = _rules_to_df(rules=rules)
+        if len(ruleset_names) > 1:
```
Comment (jaidisido) [inline awswrangler/data_quality/_get.py:44]: I think a column with the ruleset name should only be added if there is multiple ones. But let me know if you disagree

Comment (LeonLuttenberger) [reply awswrangler/data_quality/_get.py:44]: What's the major downside to adding the ruleset name always? I like the idea of the structure of the DataFrame to be consistent (i.e. it should either always have `ruleset`).

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 605be515c1373202a47bddad3d29c0f430a2abbb
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bbcf8a08-af21-48be-823b-b4df7f0beb10%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 82cd5555c7e711df3cd3f8095fadb4c3066997cc
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7e90438d-eaa2-4e1f-a634-7f9e82cbcecc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/data_quality/_create.py (hunk 1 above)
Comment (kukushking) [reply awswrangler/data_quality/_create.py:140]: How about append?

**Pull request merged**